python/lsst/ts/externalscripts/auxtel/latiss_wep_align.py

```diff
@@ -0,0 +1,286 @@
+
+        Returns
+        -------
+        results : `LatissAlignResults`
+            A dataclass containing the results of the calculation.
+        """
+
+        loop = asyncio.get_running_loop()
+
+        with concurrent.futures.ProcessPoolExecutor(max_workers=1) as pool:
+
+            self.log.debug(
+                "Running wep with: "
+                f"intra_visit_id={self.intra_visit_id}, "
+                f"extra_visit_id={self.extra_visit_id}, "
+                f"donut_diameter={2 * self.side}, "
+                f"timeout_get_image={self.timeout_get_image}. "
+            )
+
+            (
+                self.intra_result,
+                self.extra_result,
+                wep_results,
+            ) = await loop.run_in_executor(
+                pool,
+                functools.partial(
+                    run_wep,
+                    self.intra_visit_id,
+                    self.extra_visit_id,
+                    2 * self.side,
+                    self.timeout_get_image,
+                ),
+            )
+
+            # output from wep are in microns, need to convert to nm.
+            self.zern = [
+                -wep_results.outputZernikesAvg[0][4] * 1e3,
+                wep_results.outputZernikesAvg[0][3] * 1e3,
+                wep_results.outputZernikesAvg[0][0] * 1e3,
+            ]
+
+        return self.calculate_results()
+
+
+def run_wep(
+    intra_visit_id: int,
+    extra_visit_id: int,
+    donut_diameter: int,
+    timeout_get_image: float,
+) -> typing.Tuple[Struct, Struct, Struct]:
+
+    best_effort_isr = BestEffortIsr()
+
+    exposure_intra = get_image(
+        parse_visit_id(intra_visit_id),
+        best_effort_isr,
+        timeout=timeout_get_image,
+    )
+    exposure_extra = get_image(
+        parse_visit_id(extra_visit_id),
+        best_effort_isr,
+        timeout=timeout_get_image,
+    )
+
+    quick_frame_measurement_config = QuickFrameMeasurementTask.ConfigClass()
+    quick_frame_measurement_task = QuickFrameMeasurementTask(
+        config=quick_frame_measurement_config
+    )
+
+    result_intra = quick_frame_measurement_task.run(
+        exposure_intra, donutDiameter=donut_diameter
+    )
+    result_extra = quick_frame_measurement_task.run(
+        exposure_extra, donutDiameter=donut_diameter
+    )
+
+    if not result_intra.success or not result_extra.success:
+        raise RuntimeError(
+            f"Centroid finding algorithm was unsuccessful. "
+            f"Intra image ({exposure_intra}) success is {result_intra.success}. "
+            f"Extra image ({exposure_extra}) success is {result_extra.success}."
+        )
+
+    dy = (
+        result_extra.brightestObjCentroidCofM[0]
+        - result_intra.brightestObjCentroidCofM[0]
+    )
+    dx = (
+        result_extra.brightestObjCentroidCofM[1]
+        - result_intra.brightestObjCentroidCofM[1]
+    )
+    dr = np.sqrt(dy**2 + dx**2)
+
+    position_out_of_range = dr > 100
```

Comment:
I don't like that this is hardcoded... I know i'm the one that did it... I have mixed feelings about making it a config parameter.

In fact, since the entire catalogue is being used (below), can we just get rid of it?

Reply:
well.. not really.. as I mentioned below (sorry I am going through these from the bottom up), I am still using best effort isr and quick frame measurement to pre-process the data and find the donuts so, in a sense, this is still needed. 

I think we can have a conversation about whether we should stop using best effort  isr and quick frame measurement, but we spend quite some time improving these so, I think we should continue using them.

Reply:
Ok. Let's have a chat but we can make changes under a different PR.